### ISSUE TITLE
Make filter queries more intuitive

### ISSUE THREAD

**Issue by AideeAmei** [CleverRaven/Cataclysm-DDA#78393]
### Is your feature request related to a problem? Please describe.

It seems to me not all filters work the same and it's not possible to use some basic operators like AND, OR, NOT. Say I want to cook a food from certain ingredients, it's not possible to find recipes that would use preferred ingredients, it's not even possible to exclude fish. My player character hates fish, they smell awful to them!

### Solution you would like.

Make filter queries work like we're in 2024, not 1846. Also it would be nice if they worked as similarly as possible on different screens.

### Describe alternatives you have considered.

Read recipes manually.
Cook random stuff.
Eat fish and loose morale.

### Additional context

I know there are other FRs regarding this, what this one is trying to say is more high level, unifying. I realize it's not possible to filter recipes the exact same way as items, but some basic logic should apply equally.

**Comment by moxian** [comment]
This reads like a rant.

> Say I want to cook a food from certain ingredients, it's not possible to find recipes that would use preferred ingredients

It is, if you search for `c:ingredient`.

> it's not even possible to exclude fish. My player character hates fish, they smell awful to them

if don't have fish near crafting area, then the recipes would be greyed out, and moved to the bottom of the results list where they won't bother you.

But also, you don't want to exclude something like `meat sandwich` from the search results just because it *can* use fish instead of meat.


Please write out a detailed explanation of how you think the search should work if you want this issue to be taken seriously.

**Comment by AideeAmei** [comment]
Yeah, I'm sleepy. Say I want to see recipes that need 2 ingredients, I want to chose the ingredient variant for each and then want to exclude another ingredient. Moving ingredients away each time I want to filter something is not a solution, that's a workaround and not such that would take no in-game time, which can have consequences.

Example: foods that would use beans but would also use pickled veggy - not as one interchangeable ingredient but both - and they would also not require fish. These are probably 2 separate things, but yeah, I had energy just for that.
